### ISSUE TITLE
est_herit funny behavior with missing values masking covariate

### ISSUE THREAD

**Issue by byandell** [rqtl/qtl2scan#112]
If there are missing values in pheno that lead to dependent columns in covariates, then est_herit behaves not as expected. It seems to return value 0.381966 in some cases. Example below adds column of zeroes to covar (creating covar2) except the first 2 values are non-zero. Now put NA in the first two rows of pheno. In est_herit(), neither get_common_ids() or drop_depcols() catch issue. Actual calculations are in calc_hsq_clean(), but there drop_depcols() is not called unless Xcovar is provided. Not sure what best code fix is here.
 
Consider following code:

```
library(qtl2scan)
example(est_herit)
covar2 <- cbind(a=covar,b=0)
pheno[1:2,1] <- NA
covar2[1:2,2] <- 1
hsq <- est_herit(pheno, kinship, covar)
hsq2 <- est_herit(pheno, kinship, covar2)
hsq3 <- est_herit(pheno, kinship, covar2[,-2, drop=FALSE])
result1 <- cbind(hsq,hsq2,hsq3)
pheno[1:2,1:2] <- NA
hsq <- est_herit(pheno, kinship, covar)
hsq2 <- est_herit(pheno, kinship, covar2)
hsq3 <- est_herit(pheno, kinship, covar2[,-2, drop=FALSE])
result2 <- cbind(hsq,hsq2,hsq3)
```

The results follow.

```
> result1
             hsq      hsq2      hsq3
liver  0.7038096 0.3819660 0.7038096
spleen 0.5437362 0.5452198 0.5437362
> result2
            hsq     hsq2      hsq3
liver  0.7038096 0.381966 0.7038096
spleen 0.5443932 0.381966 0.5443932
```

**Comment by byandell** [comment]
I wonder if this will also arise with scan1 or scan1coef?

**Comment by kbroman** [comment]
I'd previously fixed this problem in `scan1()` etc (PR #108). I should have realized at the time that it would also be a problem in `est_herit()`. Oops. It should be fixed now, in PR #114. 

(This was PR #113, but one of the tests failed on [Travis](https://travis-ci.org/rqtl/qtl2scan) and when I pushed another commit, for some reason GitHub wasn't adding it to the pull request as it is supposed to do, so I closed PR #113 and created a new one, PR #114.)
